### DISCUSSION TITLE
Access a selfhosted outline instance through a McAfee Web Gateway proxy not allowing websockets

### DISCUSSION THREAD

**Discussion by hilsonp** [outline/outline#6969] [Self Hosting]
Everything was fine with my selfhosted outline instance.
I could access it from my home laptop on my lan and from the internet through an nginx reverse proxy (and protected by Authelia).

The problem I face is when trying to access my outline documents from the office.
Websockets won't establish as I'm going through a McAfee Web Gateway proxy that will only this only to well known servers but not my personal server.

https://kcm.trellix.com/corporate/index?page=content&id=KB84052&locale=en_US

The IT department will never add my server in the list of exception.
What solution am I left with ?

Is it possible to disable websocket usage ? I do not care about collaboration as I'm the only one writing into this family wiki.
If it is not possible, should I seek for another wiki ? I'm really pleased with all other features provided by Outline.

Thank you for this nice solution. I'll be your advocate when I visit customers looking for a replacement to their big fat wiki ;-)

Best regards,

Pierre

**Comment by tommoor** [comment]
> Is it possible to disable websocket usage ?

It's not possible, the entire product is built around websocket connectivity I'm afraid. Why a gateway would block that specific usage is a very weird setup, there's nothing more or less secure about wss than https.

**Reply by hilsonp** [reply]
I know that with https, the corporate proxy becomes the ssl termination, unencrypts the traffic, inspects it and encrypts it again. 
Maybe this is not feasible with wss. 
The link in the first post says it all :-( 
So, I guess there is no solution for me. This too good to be true.
